kotsadm/pkg/updatechecker/updatechecker.go

```diff
@@ -205,6 +206,32 @@ func CheckForUpdates(appID string, deploy bool) (int64, error) {
 
 	// if there are updates, go routine it
 	if len(updates) == 0 {
+		if !deploy {
+			return 0, nil
+		}
+
+		// ensure that the latest version is deployed
+		allVersions, err := version.GetVersions(a.ID)
+		if err != nil {
+			return 0, errors.Wrap(err, "failed to list app versions")
+		}
+
+		// get the first version, the array must contain versions at this point
+		// this function can't run without an app
+		latestVersion := allVersions[len(allVersions)-1]
```

Comment:
judging by the implementation for `GetVersions`, this could be an empty array

Reply:
i'll add a check here.  my comment was more about "many many other places above me in the call stack would have failed, and it's impossible to reach this point".  but fair, i'll add